var/spack/repos/builtin/packages/libxsmm/package.py

```diff
@@ -60,6 +61,8 @@ class Libxsmm(Package):
             description='Unoptimized with call-trace (LIBXSMM_TRACE).')
     variant('header-only', default=False,
             description='Produce header-only installation')
+    # variant +header-only is only available since v1.6.2
+    conflicts('+header-only', when='@:1.6.2')
```

Comment:
`conflicts` actually has an optional `msg` attribute that you can set to this comment if you want.